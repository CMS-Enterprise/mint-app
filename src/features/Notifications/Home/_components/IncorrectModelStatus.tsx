import React from 'react';
import { Trans, useTranslation } from 'react-i18next';
import { Grid } from '@trussworks/react-uswds';
import { IncorrectModelStatusActivityMeta } from 'gql/generated/graphql';

import UswdsReactLink from 'components/LinkWrapper';
import PageHeading from 'components/PageHeading';

const IncorrectModelStatus = ({
  currentStatus,
  modelPlan,
  modelPlanID,
  phaseSuggestion
}: IncorrectModelStatusActivityMeta) => {
  const { t: notificationsT } = useTranslation('notifications');

  return (
    <Grid
      desktop={{ col: 12 }}
      className="border-1 border-base-lightest padding-x-3 padding-y-5"
      data-testid="incorrect-model-status"
    >
      <PageHeading headingLevel="h2" className="margin-top-0 margin-bottom-2">
        {notificationsT('index.incorrectModelStatus.heading', {
          modelName: modelPlan?.modelName
        })}
      </PageHeading>
      <p className="margin-top-0 margin-bottom-3">
        {notificationsT('index.incorrectModelStatus.subheading')}
      </p>

      <div>
        <p className="margin-bottom-1">
          <Trans
            i18nKey="notifications:index:incorrectModelStatus.currentStatus"
            components={{ bold: <strong /> }}
            values={{ currentStatus }}
          />
        </p>
        <p className="margin-top-1">
          <Trans
            i18nKey="notifications:index:incorrectModelStatus.newStatus"
            components={{ bold: <strong /> }}
            values={{ newStatus: phaseSuggestion?.suggestedStatuses[0] }}
          />
        </p>
      </div>

      <UswdsReactLink
        to={`/models/${modelPlanID}/collaboration-area/status?model-status=${phaseSuggestion?.suggestedStatuses[0]}`}
        className="display-block deep-underline margin-y-4"
      >
        {notificationsT('index.incorrectModelStatus.cta')}
      </UswdsReactLink>

      <p>
        <Trans
          i18nKey="notifications:index:incorrectModelStatus.adjustTimeline"
          components={{
            link1: (
              <UswdsReactLink
                className="deep-underline"
                to={`/models/${modelPlanID}/collaboration-area/model-timeline`}
              />
            )
          }}
        />
      </p>
    </Grid>
  );
};

export default IncorrectModelStatus;
