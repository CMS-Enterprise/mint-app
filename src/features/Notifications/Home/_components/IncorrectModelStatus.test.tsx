import React from 'react';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import { render, screen, waitFor } from '@testing-library/react';
import { ModelPhase, ModelStatus } from 'gql/generated/graphql';

import IncorrectModelStatus from './IncorrectModelStatus';

const incorrectModelStatusProps = {
  modelPlanName: 'Wonderful Plan',
  modelPlanID: 'notarealid',
  modelPlan: {
    id: 'notarealid',
    modelName: 'Wonderful Plan',
    status: ModelStatus.CMS_CLEARANCE
  },
  phaseSuggestion: {
    __typename: 'PhaseSuggestion' as const,
    phase: ModelPhase.IN_CLEARANCE,
    suggestedStatuses: [ModelStatus.ENDED]
  }
};

describe('Incorrect Model Status in Notifications', () => {
  it('should render accordingly and matches snapshot', async () => {
    const router = createMemoryRouter(
      [
        {
          path: '/notifications',
          element: <IncorrectModelStatus {...incorrectModelStatusProps} />
        }
      ],
      {
        initialEntries: [`/notifications`]
      }
    );

    const { asFragment } = render(
      <RouterProvider router={router} addTypename={false} />
    );

    await waitFor(() => {
      expect(screen.getByTestId('incorrect-model-status')).toBeInTheDocument();
      expect(screen.getByText(/Wonderful Plan/i)).toBeInTheDocument();
      expect(
        screen.getByText(/timeline suggests that it is now in clearance/i)
      ).toBeInTheDocument();
      expect(screen.getByText(/In CMS clearance/i)).toBeInTheDocument();

      expect(
        screen.getByText('Yes, update my modelâ€™s status').closest('a')
      ).toHaveAttribute(
        'href',
        `/models/${incorrectModelStatusProps.modelPlanID}/collaboration-area/status?model-status=${incorrectModelStatusProps.phaseSuggestion.suggestedStatuses[0]}`
      );
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
