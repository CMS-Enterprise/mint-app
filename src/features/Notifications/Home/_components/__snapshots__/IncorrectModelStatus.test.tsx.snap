// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Incorrect Model Status in Notifications > should render accordingly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="desktop:grid-col-12 border-1 border-base-lightest padding-x-3 padding-y-5 line-height-body-5"
    data-testid="incorrect-model-status"
  >
    <h2
      aria-live="polite"
      class="mint-h1 margin-top-0 margin-bottom-2"
      tabindex="-1"
    >
      It's time to update the model status for Wonderful Plan
    </h2>
    <p
      class="margin-top-0 margin-bottom-3"
    >
      Your model’s anticipated timeline suggests that it is now in clearance. Would you like to update the status of your model to reflect that? If your model is not yet in clearance, please adjust your model’s anticipated timeline.
    </p>
    <div>
      <p
        class="margin-bottom-1"
      >
        <strong>
          Current status:
        </strong>
         In CMS clearance
      </p>
      <p
        class="margin-top-1"
      >
        <strong>
          New status:
        </strong>
         Ended
      </p>
    </div>
    <a
      class="usa-link display-block deep-underline margin-y-4"
      data-discover="true"
      href="/models/notarealid/collaboration-area/status?model-status=ENDED"
    >
      Yes, update my model’s status
    </a>
    <p>
      If you’re not ready to update your model’s status, please 
      <a
        class="usa-link deep-underline"
        data-discover="true"
        href="/models/notarealid/collaboration-area/model-timeline"
      >
        adjust your model’s anticipated timeline.
      </a>
    </p>
  </div>
</DocumentFragment>
`;
